src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -101,22 +102,8 @@ final class RiskProvider: RiskProviding {
 			return
 		}
 
-		guard store.lastSuccessfulSubmitDiagnosisKeyTimestamp == nil else {
-			Log.info("RiskProvider: Keys were already submitted. Don't start new risk detection.", log: .riskDetection)
-
-			// Keep downloading key packages and trace warning packages for plausible deniability
-
-			downloadKeyPackages()
-
-			appConfigurationProvider.appConfiguration().sink { [weak self] appConfiguration in
-				self?.downloadTraceWarningPackages(with: appConfiguration, completion: { _ in })
-			}.store(in: &subscriptions)
-
-			return
-		}
-
-		guard !WarnOthersReminder(store: store).positiveTestResultWasShown else {
-			Log.info("RiskProvider: Positive test result was already shown. Don't start new risk detection.", log: .riskDetection)
+		guard !coronaTestService.hasAtLeastOneShownPositiveOrSubmittedTest else {
```

Comment:
The downloads should not run anymore if there is a positive or submitted test?

Reply:
Once at least one registered test has a positive result that was shown to the user or the keys were submitted from it, only the downloads should run. No risk calculation should happen, as we don't show the risk card on the Home Screen in that state. Basically the same logic as we had before, just applying to both tests.